src/shadowbox/infrastructure/prometheus_scraper.ts

```diff
@@ -92,7 +93,29 @@ export function runPrometheusScraper(
       // TODO(fortuna): Consider saving the output and expose it through the manager service.
       runProcess.stdout.pipe(process.stdout);
       runProcess.stderr.pipe(process.stderr);
+
+      await waitForPrometheusReady(prometheusEndpoint);
       resolve(runProcess);
     });
   });
 }
+
+async function waitForPrometheusReady(prometheusEndpoint: string) {
+  logging.debug('Waiting for Prometheus to be ready...');
+  let ready = false;
+  while (!ready) {
+    ready = await isPrometheusReady(prometheusEndpoint);
+  }
+  logging.debug('Prometheus is ready');
+}
+
+function isPrometheusReady(prometheusEndpoint: string): Promise<boolean> {
```

Comment:
This is not Prometheus-specific. Maybe rename to something like "isHttpEndpointHealthy"?

Reply:
Done.